src/native/libs/System.Security.Cryptography.Native.Apple/pal_swiftbindings.swift

```diff
@@ -86,3 +85,85 @@ public func AppleCryptoNative_ChaCha20Poly1305Decrypt(
         return 0
     }
 }
+
+@_cdecl("AppleCryptoNative_AesGcmEncrypt")
+public func AppleCryptoNative_AesGcmEncrypt(
+    keyPtr: UnsafeMutableRawPointer,
+    keyLength: Int32,
+    noncePtr: UnsafeMutableRawPointer,
+    nonceLength: Int32,
+    plaintextPtr: UnsafeMutableRawPointer,
+    plaintextLength: Int32,
+    ciphertextBuffer: UnsafeMutablePointer<UInt8>,
+    ciphertextBufferLength: Int32,
+    tagBuffer: UnsafeMutablePointer<UInt8>,
+    tagBufferLength: Int32,
+    aadPtr: UnsafeMutableRawPointer,
+    aadLength: Int32
+ ) -> Int32 {
+    let nonceData = Data(bytesNoCopy: noncePtr, count: Int(nonceLength), deallocator: Data.Deallocator.none)
+    let key = Data(bytesNoCopy: keyPtr, count: Int(keyLength), deallocator: Data.Deallocator.none)
+    let plaintext = Data(bytesNoCopy: plaintextPtr, count: Int(plaintextLength), deallocator: Data.Deallocator.none)
+    let aad = Data(bytesNoCopy: aadPtr, count: Int(aadLength), deallocator: Data.Deallocator.none)
+    let symmetricKey = SymmetricKey(data: key)
+
+    guard let nonce = try? AES.GCM.Nonce(data: nonceData) else {
+        return 0
+    }
+
+    guard let result = try? AES.GCM.seal(plaintext, using: symmetricKey, nonce: nonce, authenticating: aad) else {
+        return 0
+    }
+
+    assert(ciphertextBufferLength >= result.ciphertext.count)
+    assert(tagBufferLength >= result.tag.count)
+
+    result.ciphertext.copyBytes(to: ciphertextBuffer, count: result.ciphertext.count)
+    result.tag.copyBytes(to: tagBuffer, count: result.tag.count)
+    return 1
+ }
+
+@_cdecl("AppleCryptoNative_AesGcmDecrypt")
+public func AppleCryptoNative_AesGcmDecrypt(
+    keyPtr: UnsafeMutableRawPointer,
+    keyLength: Int32,
+    noncePtr: UnsafeMutableRawPointer,
+    nonceLength: Int32,
+    ciphertextPtr: UnsafeMutableRawPointer,
+    ciphertextLength: Int32,
+    tagPtr: UnsafeMutableRawPointer,
+    tagLength: Int32,
+    plaintextBuffer: UnsafeMutablePointer<UInt8>,
+    plaintextBufferLength: Int32,
+    aadPtr: UnsafeMutableRawPointer,
+    aadLength: Int32
+) -> Int32 {
+    let nonceData = Data(bytesNoCopy: noncePtr, count: Int(nonceLength), deallocator: Data.Deallocator.none)
+    let key = Data(bytesNoCopy: keyPtr, count: Int(keyLength), deallocator: Data.Deallocator.none)
+    let ciphertext = Data(bytesNoCopy: ciphertextPtr, count: Int(ciphertextLength), deallocator: Data.Deallocator.none)
+    let aad = Data(bytesNoCopy: aadPtr, count: Int(aadLength), deallocator: Data.Deallocator.none)
+    let tag = Data(bytesNoCopy: tagPtr, count: Int(tagLength), deallocator: Data.Deallocator.none)
+    let symmetricKey = SymmetricKey(data: key)
+
+    guard let nonce = try? AES.GCM.Nonce(data: nonceData) else {
+        return 0
+    }
+
+    guard let sealedBoxRestored = try? AES.GCM.SealedBox(nonce: nonce, ciphertext: ciphertext, tag: tag) else {
```

Comment:
Why "Restored" in the variable name?

Reply:
\> Because that's what it's called for the ChaCha implementation
\< Why is it called that for the ChaCha implementation?
\> I don't know. The code for this is actually years old from a few spikes trying to get this working.
\< Can we rename it then? "Restored" doesn't add a lot of value to the variable name.
\> Yeah.